### PR TITLE
Use pagelets to extend rex

### PR DESCRIPTION
**Pull request by fellipeh**
Signed-off-by: Fellipe Henrique <fpedrosa@redhat.com>

### CODE REVIEW

File: app/views/job_invocations/_leapp_tab.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<% content_for(:javascripts) do %>
+    <%= webpacked_plugins_js_for :foreman_leapp %>
+  <% end %>
+  <% content_for(:stylesheets) do %>
+    <%= webpacked_plugins_css_for :foreman_leapp %>
+  <% end %>
+
+<%= react_component('JobInvocationLeappTab', { :job_invocation_id => @job_invocation.id })%>
```
Comment (xprazak2) [inline app/views/job_invocations/_leapp_tab.html.erb]: I think we usually have newline at the end, rubocop should flag this. Could you add those newlines to other files as well please?

File: lib/foreman_leapp/engine.rb [hunk 2]
```diff
@@ -23,31 +23,27 @@ class Engine < ::Rails::Engine
...
+        end
       end
     end
 
     # Include concerns in this config.to_prepare block
     config.to_prepare do
       begin
         Host::Managed.include ForemanLeapp::HostExtensions
-        HostsHelper.prepend ForemanLeapp::HostsHelperExtensions
+        HostsHelper.include ForemanLeapp::HostsHelperExtensions
```
Comment (xprazak2) [inline lib/foreman_leapp/engine.rb]: I do not see any changes in `ForeamnLeapp::HostsHelperExtensions`, why changing it to `include`?

File: lib/foreman_leapp/engine.rb [hunk 3]
```diff
@@ -23,31 +23,27 @@ class Engine < ::Rails::Engine
...
       rescue StandardError => e
         Rails.logger.warn "ForemanLeapp: skipping engine hook (#{e})"
       end
-
-      RemoteExecutionFeature.register(
```
Comment (xprazak2) [inline lib/foreman_leapp/engine.rb]: I do not think we want to remove these.

File: webpack/index.js [hunk 4]
```diff
@@ -1 +1,23 @@
...
+/* eslint-disable import/no-unresolved */
+
 import componentRegistry from 'foremanReact/components/componentRegistry';
+
+import JobInvocationLeappTab from './components/JobInvocationLeappTab';
+// cant resolve this for now
+// import Application from './containers/Application/index';
+//import './redux';
+// Not currently mocking anything
+// import './services/api/setupMocks';
```
Comment (xprazak2) [inline webpack/index.js]: Could you remove the commented code please? We can add those imports when we have things to import.

File: webpack/index.js [hunk 5]
```diff
@@ -1 +1,23 @@
...
+
+componentRegistry.register({
+  name: 'JobInvocationLeappTab',
+  type: JobInvocationLeappTab,
+});
+
+// componentRegistry.register({
```
Comment (xprazak2) [inline webpack/index.js]: You can remove this as well.

**Comment by xprazak2** [comment]
Awesome!

I added a couple of comments, should be easy to fix.

**Comment by xprazak2** [comment]
It seem like other changes were pulled into the branch as well. I think a rebase of the commit adding pagelets might resolve this.

**Comment by adamruzicka** [comment]
Needs a rebase

Also it seems to still include changes from different PRs

**Comment by fellipeh** [comment]
I believe all changes in these PR includes chagnes from other PRs, already merged... so, closing...

**Pull request closed without merging**